otherlibs/unix/getgr.c

```diff
@@ -28,7 +28,8 @@ static value alloc_group_entry(struct group *entry)
 
   Begin_roots3 (name, pass, mem);
     name = caml_copy_string(entry->gr_name);
-    pass = caml_copy_string(entry->gr_passwd);
+    /* on some platforms, namely Android, gr_passwd can be NULL - hence this workaround */
+    pass = caml_copy_string(entry->gr_passwd ? entry->gr_passwd : "");
```

Comment:
FTR I believe this concerns a very small fraction of users given that : 
1. all modern OS do not store password (hash) in /etc/passwd, so programs do not expect gr_passwd to contain anything useful
2. current behaviour in case of NULL is a crash, so this change makes the program go further (worst case - try login with empty string as a password and fail explicitly)